### ISSUE TITLE
Add transitEncryption and authorizationConfig to efs_volume_configuration in ECS task definition 

### ISSUE THREAD

**Issue by aliusmiles** [terraform-providers/terraform-provider-aws#12810]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* resource: aws_ecs_task_definition

### Potential Terraform Configuration

```hcl
resource "aws_ecs_task_definition" "service" {
  family                = "service"
  container_definitions = "${file("task-definitions/service.json")}"

  volume {
    name = "service-storage"

    efs_volume_configuration {
      file_system_id  = aws_efs_file_system.fs.id
      root_directory = "/opt/data"

      transit_encryption_enabled = true
      transit_encryption_port    = 2049

      authorization_config {
         access_point_id = "fsap-1234"
         iam_enabled     = true
      }
    }
  }
}
```

### References

AWS Docs:
* https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_EFSVolumeConfiguration.html
* https://docs.aws.amazon.com/AmazonECS/latest/userguide/efs-volumes.html (note: json example is messed up there!)

Related issues:
#12118 not strictly related, but would be nice to use together with this one once both are implemented.

**Comment by DrFaust92** [comment]
I have this in a branch and waiting on #11965 to be merged to push it.

**Comment by DrFaust92** [comment]
depends on #12715

**Comment by jdheyburn** [comment]
Duplicate of #12809

**Comment by Jukie** [comment]
Once #11965 is merged I'll push code for authorization_config. #13136 adds transit encryption though.

**Comment by aliusmiles** [comment]
@Jukie Thanks a lot!

Closing this ticket as duplicate, let’s use #12809 for tracking.
